File: pdu/python/std_msgs/pdu_cdr_conv_Char.py
from .pdu_pytype_Char import Char
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Char(writer: CdrWriter, src: Char):
    writer.write_uint8(src.data)


def cdr_body_to_py_Char(reader: CdrReader, dst: Char):
    dst.data = reader.read_uint8()
    return dst


def py_to_cdr_Char(src: Char) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Char(writer, src)
    return writer.bytes()


def cdr_to_py_Char(cdr_payload) -> Char:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Char()
    return cdr_body_to_py_Char(reader, dst)

File: pdu/javascript/std_msgs/pdu_cdr_conv_Char.js
import { Char } from './pdu_jstype_Char.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduCharConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Char } src
     */
    static to_cdr_body(writer, src) {
        writer.write_uint8(src.data);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Char } dst
     * @returns { Char }
     */
    static cdr_body_to_js(reader, dst) {
        dst.data = reader.read_uint8();
        return dst;
    }

    /**
     * @param { Char } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Char }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Char());
    }
}
